pyProbeParticle/basUtils.py

```diff
@@ -1,131 +1,124 @@
-                    enames.append( elements.ELEMENTS[iz] )
-                except:
-                    ename = wds[0]
-                    enames.append( ename )
-                    Zs    .append( elements.ELEMENT_DICT[ename][0] )
-                try:
-                    q = float(wds[4])
-                except:
-                    q = 0
-                qs.append(q)
-            except:
-                print("cannot interpet line: ", line)
-                continue
-    xyzs = np.array( xyzs )
-    Zs   = np.array( Zs, dtype=np.int32 )
-    qs   = np.array(qs)
-    return xyzs,Zs,enames,qs
-
-def loadAtomsLines( lines ):
-    xyzs   = [] 
-    Zs     = []
-    enames = []
-    qs     = []
-    for line in lines:
-        wds = line.split()
-        try:
-            xyzs.append( ( float(wds[1]), float(wds[2]), float(wds[3]) ) )
-            try:
-                iz    = int(wds[0]) 
-                Zs    .append(iz)
-                enames.append( elements.ELEMENTS[iz] )
-            except:
-                ename = wds[0]
-                enames.append( ename )
-                Zs    .append( elements.ELEMENT_DICT[ename][0] )
-            try:
-                q = float(wds[4])
-            except:
-                q = 0.0
-            qs.append(q)
-        except:
-            if(verbose>0):
-                print("cannot interpet line: ", line)
-            continue
-    xyzs = np.array( xyzs )
-    Zs   = np.array( Zs, dtype=np.int32 )
-    qs   = np.array(qs)
-    return xyzs,Zs,enames,qs
+                Zs.append(Z)
+                extra_cols.append(wds[4:])
+            except (ValueError, IndexError):
+                raise ValueError(f'Could not interpret line in xyz file: `{line}`')
+
+    xyzs = np.array(xyzs, dtype=np.float64)
+    Zs = np.array(Zs, dtype=np.int32)
+        
+    if len(extra_cols[0]) > 0:
+        qs = _getCharges(comment, extra_cols)
+    else:
+        qs = np.zeros(len(Zs), dtype=np.float64)
+
+    return xyzs, Zs, qs, comment
+
+def _getCharges(comment, extra_cols):
+    match = re.match('.*Properties=(\S*) ', comment)
+    if match:
+        # ASE format, check if one of the columns has charges
+        props = match.group(1).split(':')[6:] # [6:] is for skipping over elements and positions
+        col = 0
+        for name, size in zip(props[::3], props[2::3]):
+            if name in ['charge', 'initial_charges']:
+                qs = np.array([float(ex[col]) for ex in extra_cols], dtype=np.float64)
+                break
+            col += int(size)
+        else:
+            qs = np.zeros(len(extra_cols), dtype=np.float64)
+    else:
+        # Not ASE format, so just take first column
+        qs = np.array([float(ex[0]) for ex in extra_cols], dtype=np.float64)
+        if _notActuallyCharge(qs):
+            # qs is not actually charges based on some heuristics
+            qs = np.zeros(len(qs), dtype=np.float64)
+    return qs
+
+def _notActuallyCharge(qs):
```

Comment:
I don't think the utility functions should check physics. This sort of goes against the separation of concern principle.

Reply:
Well, you never have systems, which would have abs(charges) higher than 3, 5 or 10. You would get completely crazy results and this could help you to debug the problem. I would higher the treshold for 10 here. Maybe @ProkopHapala you could also tell your opinion?

Reply:
A better place for this check could be in `loadGeometry` after reading the xyz file?

Reply:
> Well, you never have systems, which would have abs(charges) higher than 3, 5 or 10. You would get completely crazy results and this could help you to debug the problem.

I am not questioning the need for such checks. They can be helpful. However, I would advocate for a separate module for such analysis.

My point is: why XYZ reader should be so complex? One day we want to replace the reader with something else, we need to monkey-patch all these checks. Why not already have a "check_physics" module that acts on ready-to-be-simulated structure? It could even be that I have an XYZ file with random charges that I am not even planning to use at all.

> A better place for this check could be in `loadGeometry` after reading the xyz file?

I would advocate for a separate module `check_physics`. It could even be kept inside utils - I don't have an opinion here.

Reply:
> I would advocate for a separate module `check_physics`

That would make sense. But I think this would require more discussion about where should we use it and also if there are other checks that we want to make. I propose that we just delete the check from this PR and make a new issue to discuss it further, so that this PR can be merged.

Reply:
Ok, @yakutovicha - I got your point. @NikoOinonen - I agree - let's remove it now and put it as a separate function or class in the future.
I would then propose to have the check on point-charges somewhere in the _generateElFF_ ... part. So one do not follow the calculation with really wrong system.

Reply:
Ok, I now removed the charge check, and made a new issue at #51.

Reply:
👍